### ISSUE TITLE
Plot label None in line plot

### ISSUE THREAD

**Issue by micramm** [pydata/pandas#8905]
The plot label appears when plotting a dataframe as a scatter plot but not as a line plot. Is this expected?

```python
df = pd.DataFrame(np.random.rand(2, 4), columns=['a', 'b', 'c', 'd'])
ax = df.plot(kind = 'scatter', x='a', y='b',color='DarkBlue', label='Label')
ax = df.plot(kind = 'line', x='a', y='b',color='DarkBlue', label='Label')
```
![image](https://cloud.githubusercontent.com/assets/885310/5212582/d59c7ec6-75b0-11e4-8f68-33a2b507e953.png)
![image](https://cloud.githubusercontent.com/assets/885310/5212583/db1533b6-75b0-11e4-9c77-b8e34bac6abd.png)

```
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.8.final.0
python-bits: 64
OS: Darwin
OS-release: 14.0.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.15.1
nose: 1.3.1
Cython: 0.21.1
numpy: 1.9.1
scipy: 0.14.0
statsmodels: None
IPython: 2.3.1
sphinx: 1.2.3
patsy: None
dateutil: 2.2
pytz: 2014.9
bottleneck: 0.8.0
tables: 3.1.1
numexpr: 2.4
matplotlib: 1.4.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
rpy2: None
sqlalchemy: None
pymysql: None
psycopg2: None
```

**Comment by petrushy** [comment]
in 0.15.2 release a quick fix seems to use data.plot(y=['col']) instead of data.plot(y='col')

**Comment by jorisvandenbossche** [comment]
This was also closed by #9574
